aws-android-sdk-appsync-runtime/src/main/java/com/apollographql/apollo/internal/RealAppSyncSubscriptionCall.java

```diff
@@ -68,23 +68,22 @@ public void run() {
                 userCallback = callback;
                 subscriptionManager.addListener(subscription, callback);
 
-                //Ensure that the call is only made once.
-                synchronized (this) {
-                    switch (state.get()) {
-                        case IDLE: {
-                            state.set(ACTIVE);
-                            break;
-                        }
+                switch (state.get()) {
+                    case IDLE: {
+                        state.set(ACTIVE);
+                        break;
+                    }
 
-                        case CANCELED:
-                            throw new RuntimeException("Cancelled", new ApolloCanceledException("Call is cancelled."));
+                    case CANCELED:
+                        callback.onFailure(new ApolloCanceledException("Call is cancelled."));
+                        break;
 
-                        case ACTIVE:
-                            throw new IllegalStateException("Already Executed");
+                    case ACTIVE:
+                        callback.onFailure(new ApolloException("Already Executed."));
```

Comment:
Minor. Let's remove the trailing period '.' to match with the existing message exactly.